### PR TITLE
[CodeGenLib] Typealias translator configuration for client and server

### PR DESCRIPTION
**Pull request by stefanadranca**
Typealias translator configuration for client and server type aliases

Motivation:

The typealias translator should generate typealiases for the client and server protocols only if the associated code (client/server code) will be generated. Also, at the moment the typealias translator doesn't generate the `ClientProtocol` typealias.

Modifications:

- added a new initialiser for the `TypealiasTranslator` that sets the client and server property.
- Based on the client and server properties' values, the typealias translator generates or not the protocol typealiases.
- created method for client protocol typalias generation.
- modified tests, so they include the client code.
- added new tests for generating only client or server code.

Result:

The typealias translator will generate typealiases for client and server protocols only when the associated code is also generated, getting rid of the possibility of an error (when the protocol doesn't exist).

### CODE REVIEW

File: Sources/GRPCCodeGen/Internal/Translator/TypealiasTranslator.swift [hunk 1]
```diff
@@ -307,4 +323,19 @@ extension TypealiasTranslator {
...
   }
+
+  private func makeClientProtocolTypealias(
+    for service: CodeGenerationRequest.ServiceDescriptor
+  ) -> Declaration {
+    let namespacedPrefix: String
+    if service.namespace.isEmpty {
+      namespacedPrefix = service.name
+    } else {
+      namespacedPrefix = "\(service.namespace)_\(service.name)"
```
Comment (stefanadranca) [inline Sources/GRPCCodeGen/Internal/Translator/TypealiasTranslator.swift:334]: After merging the server code translator PR, I will use the service property from the extension to get the namespacedPrefix

File: Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift [hunk 2]
```diff
@@ -58,13 +58,122 @@ final class SnippetBasedTranslatorTests: XCTestCase {
...
+    )
+    let expectedSwift =
+      """
+      enum namespaceA {
+          enum ServiceA {
+              enum Methods {}
+              static let methods: [MethodDescriptor] = []
+              typealias StreamingServiceProtocol = namespaceA_ServiceAServiceStreamingProtocol
+              typealias ServiceProtocol = namespaceA_ServiceAServiceProtocol
+              typealias ClientProtocol = namespaceA_ServiceAClientProtocol
```
Comment (glbrntt) [inline Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift:89]: We can add a `typealias` for the `Client` as well

**Review by glbrntt — CHANGES_REQUESTED** [review]

File: Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift (hunk 2 above)
Comment (stefanadranca) [reply Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift:89]: for the struct?

Comment (glbrntt) [reply Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift:89]: yes

**Review by glbrntt — APPROVED** [review]

**Pull request merged**